### PR TITLE
Implement color/alpha modifiers

### PR DESCRIPTION
**Pull request by bunnei**
Implements the other color and alpha modifiers (several games spammed the logger with GPU errors regarding these). E.g., the Cubic Ninja intro logos now both show up with proper fades:

### CODE REVIEW

**Comment by purpasmart96** [comment]
Dun dun dun...

File: src/video_core/math.h [hunk 1]
```diff
@@ -461,23 +461,33 @@ class Vec4
     // e.g. Vec2 uv() { return Vec2(x,y); }
     // _DEFINE_SWIZZLER2 defines a single such function, DEFINE_SWIZZLER2 defines all of them for all component names (x<->r) and permutations (xy<->yx)
```
Comment (Subv) [inline src/video_core/math.h]: Please update the comment, DEFINE_SWIZZLER2 doesn't seem to exist anymore

File: src/video_core/pica.h [hunk 2]
```diff
@@ -233,19 +233,27 @@ struct Regs {
...
-            // Other values seem to be non-standard extensions
+            SourceColor         = 0x0,
+            OneMinusSourceColor = 0x1,
+            SourceAlpha         = 0x2,
+            OneMinusSourceAlpha = 0x3,
+            SourceRed           = 0x4,
+            OneMinusSourceRed   = 0x5,
+            SourceGreen         = 0x8,
+            OneMinusSourceGreen = 0x9,
+            SourceBlue          = 0xc,
```
Comment (neobrain) [inline src/video_core/pica.h]: Blank line here, please

File: src/video_core/rasterizer.cpp [hunk 3]
```diff
@@ -324,27 +296,56 @@ void ProcessTriangle(const VertexShader::OutputVertex& v0,
...
                     case ColorModifier::SourceAlpha:
-                        return { values.a(), values.a(), values.a() };
+                        return values.aaa();
 
-                    default:
-                        LOG_ERROR(HW_GPU, "Unknown color factor %d\n", (int)factor);
-                        _dbg_assert_(HW_GPU, 0);
-                        return {};
+                    case ColorModifier::OneMinusSourceAlpha:
+                        return (Math::Vec3<u8>(255, 255, 255) - values.aaa()).Cast<u8>();
```
Comment (neobrain) [inline src/video_core/rasterizer.cpp]: Apparently, I need to introduce a Vec1 class :)

**Comment by neobrain** [comment]
LGTM other than Subv's comment

**Comment by bunnei** [comment]
@neobrain should be good to go

**Pull request merged**